openlibrary/plugins/openlibrary/js/covers.js

```diff
@@ -56,6 +56,13 @@ function add_iframe(selector, src) {
 // covers/manage.html and covers/add.html
 export function initCoversAddManage() {
     $('#addcover-form').on('submit', function (event) {
+        const addLoadingStyling = () => {
+            const btn = $('#imageUpload');
+            btn.prop('disabled', true).html(btn.data('loading-text'));
```

Comment:
This doesn't need to be in a function; you can move these lines down to just after 83. That will also let the validation checks pass first before disabling the button.

Reply:
Sounds good to me! I'll go ahead and make the changes here, then have a look at the `val` issue tomorrow.

Reply:
Moved everything and it all seems to be working great; before I resolve this -- I wanted to double check whether the `val` fix should be done here or as a separate issue/PR. Thanks so much!

Reply:
Oh here is good! Should be a quick fix :) Note you'll have to do a rebase ontop of the lastest master since the other covers PR touches similar code :+1:

Reply:
Great! Will do!

Reply:
Done! And fixing the `val` bug made it clear that the checks to return an error if no file/url/id had been submitted - `if (file === '' && url == '' && coverid == ''` - had been failing the whole time because the coverid val if none is entered defaults to `undefined` and not `''`. So I threw in a quick fix for that as well by switching to `if (!file && !url && !coverid)`, so now it looks a bit cleaner and the correct error message shows if nothing is entered.

Reply:
Nice! Much cleaner 🥳